client/src/actions/embedding.ts

```diff
@@ -61,3 +62,39 @@ export const layoutChoiceAction: ActionCreator<
       annoMatrix,
     });
   };
+
+export const initialLayoutChoiceAction: ActionCreator<
+  ThunkAction<Promise<void>, RootState, never, Action<"set layout choice">>
+> =
+  (config: globals.Config) =>
+  async (dispatch: AppDispatch, getState: GetState): Promise<void> => {
+    const { annoMatrix: prevAnnoMatrix, obsCrossfilter: prevCrossfilter } =
+      getState();
+    const availableLayouts = selectAvailableLayouts({ prevAnnoMatrix });
+    const fallbackLayout = getBestDefaultLayout(availableLayouts);
+    const defaultLayout = config?.parameters?.default_embedding || "";
+
+    const finalLayout = availableLayouts.includes(defaultLayout)
+      ? defaultLayout
+      : fallbackLayout;
+
+    const [annoMatrix, obsCrossfilter] = await _switchEmbedding(
+      prevAnnoMatrix,
+      prevCrossfilter,
+      finalLayout
+    );
+
+    dispatch({
+      type: "set layout choice",
+      layoutChoice: finalLayout,
+      obsCrossfilter,
+      annoMatrix,
+    });
```

Comment:
This could just be a call of the above action, but I think it would be more clear if each action has its own explicit dispatch

Reply:
agreed!